tools/create_service_bundle.py

```diff
@@ -0,0 +1,568 @@
+
+    def create_tasks_nodetool_tpstats_files(self):
+        self.for_each_running_task_with_prefix(
+            "node", self.create_nodetool_tpstats_file
+        )
+
+    def create(self):
+        logger.info("Creating Cassandra bundle")
+        self.create_tasks_nodetool_status_files()
+        self.create_tasks_nodetool_tpstats_files()
+
+
+class ElasticBundle(BaseTechBundle):
+    @retrying.retry(
+        wait_fixed=DEFAULT_RETRY_WAIT,
+        stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS,
+    )
+    def task_exec(self, task_id, cmd):
+        full_cmd = " ".join(
+            [
+                "export JAVA_HOME=$(ls -d ${MESOS_SANDBOX}/jdk*/jre/) &&",
+                "export TASK_IP=$(${MESOS_SANDBOX}/bootstrap --get-task-ip) &&",
+                "ELASTICSEARCH_DIRECTORY=$(ls -d ${MESOS_SANDBOX}/elasticsearch-*/) &&",
+                cmd,
+            ]
+        )
+
+        return sdk_cmd.marathon_task_exec(task_id, "bash -c '{}'".format(full_cmd))
+
+    def create_stats_file(self, task_id):
+        command = "curl -s ${MESOS_CONTAINER_IP}:${PORT_HTTP}/_stats"
+        rc, stdout, stderr = self.task_exec(task_id, command)
+        self.write_file("elasticsearch_stats_{}.json".format(task_id), stdout)
+
+    def create_tasks_stats_files(self):
+        self.for_each_running_task_with_prefix("master", self.create_stats_file)
+
+    def create(self):
+        logger.info("Creating Elastic bundle")
+        self.create_tasks_stats_files()
+
+
+class HdfsBundle(BaseTechBundle):
+    def create(self):
+        logger.info("Creating HDFS bundle")
+
+
+class KafkaBundle(BaseTechBundle):
+    def create(self):
+        logger.info("Creating Kafka bundle")
+
+
+BASE_TECH_BUNDLE = {
+    "beta-cassandra": CassandraBundle,
+    "beta-elastic": ElasticBundle,
+    "beta-hdfs": HdfsBundle,
+    "beta-kafka": KafkaBundle,
+    "cassandra": CassandraBundle,
+    "elastic": ElasticBundle,
+    "hdfs": HdfsBundle,
+    "kafka": KafkaBundle,
+}
+
+
+class ServiceBundle(Bundle):
+    DOWNLOAD_FILES_WITH_PATTERNS = ["^stdout(\.\d+)?$", "^stderr(\.\d+)?$"]
+
+    @retrying.retry(
+        wait_fixed=DEFAULT_RETRY_WAIT,
+        stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS,
+    )
+    def install_service_cli(self):
+        sdk_cmd.run_cli(
+            "package install {} --cli --yes".format(self.package_name),
+            print_output=False,
+        )
+
+    @retrying.retry(
+        wait_fixed=DEFAULT_RETRY_WAIT,
+        stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS,
+    )
+    def create_dcos_version_file(self):
+        output = sdk_cmd.run_cli("--version", print_output=False)
+        self.write_file("dcos_version.txt", output)
+
+    @retrying.retry(
+        wait_fixed=DEFAULT_RETRY_WAIT,
+        stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS,
+    )
+    def create_task_file(self):
+        output = sdk_cmd.run_cli("task --json", print_output=False)
```

Comment:
Note: This has an implicit 100 task limit.

Reply:
(also, would this not be available in `state.json`?)

Reply:
The latest changes use Mesos state to get task information.